packages/protocol/src/peer/PeerSet.ts

```diff
@@ -358,8 +354,11 @@ export class PeerSet implements ImmutableSet<OperationalPeer>, ObservableSet<Ope
             // This ends discovery without triggering promises
             mdnsScanner?.cancelOperationalDeviceDiscovery(this.#sessions.fabricFor(address), address.nodeId, false);
         }
-        this.#clients.forEach(client => client.close());
-        this.#clients.clear();
+
+        for (const { address } of this.#peers) {
+            await this.disconnect(address, false);
+        }
+
```

Comment:
clearing this.#peers is included somewhere else?

Reply:
Might be, where do you see it?  If the peers are cleaned up elsewhere they shouldn't be in the PeerSet anymore so this doesn't preclude cleaning them up as part of another process.  This just ensures we don't leak any if we use in a context where that logic doesn't apply.

Reply:
I just asked because the old code cleared the set after having closed all entries, the new code does not

Reply:
Ah, got it.  `this.#clients` was cleared previously but should have been removed as it's owned by `InteractionClientProvider` now.  PeerSet just creates an `ExchangeProvider` which encapsulates connection management.

Reply:
The point of this BTW was to decouple `PeerSet` from `InteractionClient` so I could use an alternate interaction implementation.